kaggleの住宅価格予想のコンペなどで上位スコアの方のコードをまねて
相関係数を求めるコードを書いたらバリューエラーになり対処に時間がかかったのでメモ

mat = data_w.corr('pearson')　
と参照したコードをそのまま実行した際にバリューエラーが発生
古いバージョンは「numeric_only」の引数は書かなくてもTrueになっている為
データにobject型のデータなどは使わない為エラーにならなかった

pandasのバージョン2.0.0からDataFrame.corrメソッドの仕様が変更になり、
現在では引数numeric_onlyはデフォルトでFalseになっている。
上記が要因と判明
corr(numeric_only=True)と引数を指定して解決出来た
※相関係数は
'pearson': ピアソンの積率相関係数（デフォルト）
'kendall': ケンドールの順位相関係数
'spearman': スピアマンの順位相関係数
上記3つから指定可能
記述例:dataflame.corr('pearson', numeric_only=True)
